server/routes/api/dashboard.js

```diff
@@ -31,6 +31,40 @@ router.get('/getCoursesByStudent', (req, res) => {
     });
 });
 
+
+/**drops a section by a given student, such that it removes him from the registered_students list
+ * of a given course
+ * ==========================================
+ * Example api call body:
+ * {
+    "studentID": "5f984a44da9eb32ba01d31dd",
+    "sectionID" : "5f984a44da9eb32ba01d31df"
+    }
+ */
+router.put('/dropSection', (req, res) => {
+    let studentID = req.body.studentID;
+    let sectionID = req.body.sectionID;
+
+    Course.findOne({ _id: sectionID }, function(err, course) {
+
+        let index = course.registered_students.indexOf(studentID)
+
+        if(index == -1){
+            return res.status(500).send(err);
+        }
+        course.registered_students.splice(index, 1);
```

Comment:
Here you are removing the student from the list of registered students but you also need to remove the student from the seating_arrangement 2d array. Currently, if a student drops the couse the student would still appear to have a seat in the class. You'll have to search through the 2d array and remove it (replace with null) if you find that student's id in a seat.

Reply:
@abearcher Do you need help figuring out how to do it or did you just not have time during the past week?